### PR TITLE
Implement simple protocol fee hook

### PR DESCRIPTION
**Pull request by yorhodes**
This has a few PRs merged into it ...
https://github.com/hyperlane-xyz/cosmwasm/pull/6

### CODE REVIEW

File: contracts/hooks/fees/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+}
+
+fn get_mailbox(_deps: Deps) -> Result<MailboxResponse, ContractError> {
+    Ok(MailboxResponse {
+        mailbox: "unrestricted".to_string(),
+    })
+}
+
+fn quote_dispatch() -> Result<QuoteDispatchResponse, ContractError> {
+    Ok(QuoteDispatchResponse { fees: vec![] })
```
Comment (nambrot) [inline contracts/hooks/fees/src/lib.rs]: Doesn't this need to reference the fee in storage?

File: contracts/hooks/fees/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                metadata: HexBinary::default(),
+                message: gen_bz(100),
+            }),
+        )
+        .map_err(|e| e.to_string())
+        .unwrap();
+    }
+
+    #[rstest]
+    fn test_query(deps: TestDeps) {
```
Comment (nambrot) [inline contracts/hooks/fees/src/lib.rs]: Add test that checks for the actual fee?

**Review by byeongsu-hong — APPROVED** [review]

File: contracts/hooks/fees/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Ownable(msg) => Ok(hpl_ownable::handle(deps, env, info, msg)?),
+        ExecuteMsg::FeeHook(msg) => match msg {
+            FeeHookMsg::SetFee { fee } => {
```
Comment (byeongsu-hong) [inline contracts/hooks/fees/src/lib.rs]: it seems `SetFee` needs to check ownership

File: contracts/hooks/fees/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                }
+                .into();
+
+                Ok(Response::new()
+                    .add_message(claim_msg)
+                    .add_event(new_event("claim")))
+            }
+        },
+        ExecuteMsg::PostDispatch(_) => {
+            let fee = COIN_FEE.load(deps.storage)?;
```
Comment (byeongsu-hong) [inline contracts/hooks/fees/src/lib.rs]: Should `COIN_FEE` contains only one denom?

**Review by byeongsu-hong — CHANGES_REQUESTED** [review]

**Comment by byeongsu-hong** [comment]
Codecov finally works 😞

File: contracts/hooks/fees/src/lib.rs (hunk 4 above)
Comment (yorhodes) [reply contracts/hooks/fees/src/lib.rs]: good question, I think for now, for simplicity, yes

**Review by byeongsu-hong — APPROVED** [review]

**Pull request merged**